packages/rest/src/rest.server.ts

```diff
@@ -884,18 +884,25 @@ export class RestServer extends Context implements Server, HttpServerLike {
     this._httpServer = undefined;
   }
 
-  protected _onUnhandledError(req: Request, res: Response, err: Error) {
+  protected _onUnhandledError(
+    req: Request,
+    res: Response,
+    err: Error,
+    next: Function,
+  ) {
     if (!res.headersSent) {
-      res.statusCode = 500;
-      res.end();
+      // The error can be thrown from cors middleware
+      // See https://github.com/strongloop/loopback-next/issues/4530
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      res.status((err as any).statusCode || 500).send({error: err});
+      return;
     }
 
-    // It's the responsibility of the Sequence to handle any errors.
-    // If an unhandled error escaped, then something very wrong happened
-    // and it's best to crash the process immediately.
```

Comment:
This removed code comment is explaining how was the original HTTP server designed. IMO, the problem described by #4530 is caused by the way how we hacked CORS support into our HTTP server, ignoring the Sequence-based design.

My main objection against the proposed changes is that it creates two kinds of error responses: errors handled inside a Sequence will return different HTTP response body than errors reported by additional middleware.

For longer term, I would like us to execute all middleware (including CORS) as a new sequence action, see https://github.com/strongloop/loopback-next/issues/1293

To fix the problem described in https://github.com/strongloop/loopback-next/issues/4530, I am proposing to mount a new Express error handler middleware just after the CORS middleware.

```ts
    // Allow CORS support for all endpoints so that users
    // can test with online SwaggerUI instance
    this._expressApp.use(cors(this.config.cors));

    this._expressApp.use(function handleCorsErrors(err, request, response, next) {
      // Handle CORS-related errors.
      // First try to obtain the error handler from the application.
      // This works well if application's sequence uses SequenceActions.REJECT
      // to resolve the error handler, as applications scaffolded by lb4 do.
      // When the app is using a different approach to Sequence and/or error handling,
      // then we will probably produce a different response body, but I guess that's inevitable
      // right now.
      ctx.get(SequenceActions.REJECT, {optional: true})
        .then(reject => {
          if (reject) {
            return reject({request, response}, err)
          }
          // App does not provide any standard error handler. What to do now?
          // Maybe it's ok to crash the app?
          next(err);
          // Alternatively, we can use strong-error handler directly
          writeErrorToResponse(err, request, response);
        })
        .catch(unexpectedErr => next(unexpectedErr));
    });

    // Set up endpoints for OpenAPI spec/ui
    this._setupOpenApiSpecEndpoints();

    // etc.
```

Reply:
I generalized your idea into an Express error handler for all middleware.